### PR TITLE
SQS: fix trimming .fifo from queue name

### LINKED ISSUE #888: Improper queue name parsing for SQS FIFO queues ending with an 'f'
Due to improper use of 'rstrip' queue names ending with an 'f' will not be parsed properly
```
>>> queue_name_ending_with_an_f = 'queue_name_ending_with_an_f.fifo'
>>> queue_name_ending_with_an_f.rstrip('.fifo')
'queue_name_ending_with_an_'
>>> queue_name_ending_with_an_f.split('.')[:1][0]
'queue_name_ending_with_an_f'
>>>
```

fixed in #887

### PR DESCRIPTION
**Pull request by ikonst**
A more conservative take on https://github.com/celery/kombu/pull/887 + tests.

Fixes #888.

### CODE REVIEW

**Comment by ikonst** [comment]
cc @thedrow

**Comment by auvipy** [comment]
plz fix the flake8 error

File: kombu/transport/SQS.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def _reset_cycle(self):
     def entity_name(self, name, table=CHARS_REPLACE_TABLE):
         """Format AMQP queue name into a legal SQS queue name."""
         if name.endswith('.fifo'):
-            partial = name.rstrip('.fifo')
+            partial = name[:-len('.fifo')]
```
Comment (thedrow) [inline kombu/transport/SQS.py]: Why is this implementation better?
What was wrong with rstrip that you have to change it?

File: t/unit/transport/test_SQS.py [hunk 2]
```diff
@@ -224,6 +224,12 @@ def test_none_hostname_persists(self):
         conn = Connection(hostname=None, transport=SQS.Transport)
         assert conn.hostname == conn.clone().hostname
 
+    def test_entity_name(self):
```
Comment (thedrow) [inline t/unit/transport/test_SQS.py]: Good test!

File: kombu/transport/SQS.py (hunk 1 above)
Comment (thedrow) [reply kombu/transport/SQS.py]: Nevermind. I didn't read the PR description.

[the remark this answers is not in the document]
Comment (ikonst) [reply kombu/transport/SQS.py]: We can, but then should we also extract `'.fifo'` into a constant? Performance-wise I'm not sure it matters:
```
>>> timeit("len('.fifo')", number=1_000_000)
0.068084963000004
```
and readability wise it adds a layer of indirection. (I really wish CPython just optimized away this constant `len`.)

**Comment by thedrow** [comment]
Thank you very much!

File: kombu/transport/SQS.py (hunk 1 above)
Comment (ikonst) [reply kombu/transport/SQS.py]: `rstrip` is a trap. Not the first time I see someone stumble into it.

**Pull request merged**